### ISSUE TITLE
Integration with cVim

### ISSUE THREAD

**Issue by rafi** [deanoemcke/thegreatsuspender#271]
I'm an avid [cVim](https://github.com/1995eaton/chromium-vim) user, and I love The Great Suspender. However, on suspended tabs, cVim stops working. Is there a way I could keep using both these awesome plugins together?

**Comment by tmetten** [comment]
I was experiencing this issue myself, so I'm glad someone beat me in reporting it. I love both extensions a lot. It would be nice if they'd play nice together.

**Comment by kalbasit** [comment]
I have the same problem. @deanoemcke is it possible to whitelist extensions so we can whitelist cVim?

**Comment by deanoemcke** [comment]
Can you give me a bit more information on what is breaking in cVim (specifically)?
The problem is most likely that extensions cannot function on another extensions tab. All suspended tabs belong to the Great Suspender extension. I don't think there is a way to work around this. But I would need a more specific example to be sure.

**Comment by deanoemcke** [comment]
The comments on this PR are also quite relevant: https://github.com/deanoemcke/thegreatsuspender/pull/423

I think essentially, they implemented the problematic keyboard shortcuts inside The Great Suspender. Not a solution that I think is really correct..

**Comment by rafi** [comment]
@deanoemcke In a suspended tab, cVim doesn't work at all. You can't even use <kbd>J</kbd>/<kbd>K</kbd> to navigate through tabs, none of cVim features work.

**Comment by kalbasit** [comment]
@deanoemcke this could be solved by loading the suspended page externally (as-in accessed via regular `http`/`https`). cVim (or any other extension) does not work over `chrome-extension://`.

**Comment by madhukar93** [comment]
*bump*

**Comment by deanoemcke** [comment]
@kalbasit you are correct that the problem could be resolve by using an external web address for suspended tabs. this however makes it dependent on an external site being up (as well as having an internet connection locally).
It also reduces its performance memory-wise which is the primary use case for the extension.

Unfortunately, I cannot allow this feature request to impact the core performance of the extension :(

**Comment by kalbasit** [comment]
@deanoemcke could you at least provide an option in the settings allowing us to set a custom URL for the suspended page?

**Comment by deanoemcke** [comment]
@kalbasit this would break a lot of functionality. the contents of a suspended tab would be out of my control. it would be a pretty significant code rewrite to implement.

Perhaps you could try my other extension, The Great Discarder. This uses native chrome discarding functionality rather than redirecting to a suspended page. Perhaps it will play nicer with vimium (i haven't tried it sorry).

**Comment by dumblob** [comment]
>Perhaps you could try my other extension, The Great Discarder. This uses native chrome discarding functionality rather than redirecting to a suspended page. Perhaps it will play nicer with vimium (i haven't tried it sorry).

Unfortunately starting from Chromium 61.x there is [an issue with focusing of discarded tabs](https://github.com/deanoemcke/thegreatdiscarder/issues/17 ). Therefore it doesn't play much nicer with Vimium/cVim :cry:.

**Comment by Eduruiz** [comment]
I have a suggestion here, as a vimium user for a long time and a new user of TGS I's struggling a bit.
So basically the only things I think someone needs to use on a suspended tab is `J` and `K` to change tabs, I also notice that TGS already has some keyboard shortcuts that the user can implement (I'm using `Ctrl + s` to un/suspend tabs, It's not an easy implementation to add the 'next tab' and 'prev tab' shortcuts?

It can have a default to `J` and `K` or simply have no defaults and vimium users implement the shortcut by their own.

**Comment by dumblob** [comment]
@Eduruiz that sounds like a good compromise and I raise my hand for that.

**Comment by schlowm0** [comment]
Ony update on this?

**Comment by afedulov** [comment]
It seems that what @Eduruiz is proposing is the best compromise. +1 to it.

**Comment by abnerf** [comment]
Any solutions so far?

**Comment by krzkrzkrz** [comment]
I second @Eduruiz suggestion here. Would be great to have this option
